diff -ur cron-3.0pl1.orig-solaris/compat.c cron-3.0pl1.orig/compat.c
--- cron-3.0pl1.orig-solaris/compat.c	2009-03-12 13:30:53.000000000 +0000
+++ cron-3.0pl1.orig/compat.c	2009-03-12 15:10:44.820233050 +0000
@@ -33,6 +33,8 @@
 #endif
 #include <errno.h>
 
+#include <sys/types.h>
+#include <sys/stat.h>
 
 /* the code does not depend on any of vfork's
  * side-effects; it just uses it as a quick
@@ -85,8 +87,8 @@
 #ifdef NEED_STRCASECMP
 int
 strcasecmp(left, right)
-	char	*left;
-	char	*right;
+	const char	*left;
+	const char	*right;
 {
 	while (*left && (MkLower(*left) == MkLower(*right))) {
 		left++;
@@ -236,3 +238,13 @@
 	return putenv(tmp);	/* intentionally orphan 'tmp' storage */
 }
 #endif
+
+/* Solaris 10 is missing mkdtemp */
+char *mkdtemp(char *template)
+{
+        if (!mktemp(template) || mkdir(template, 0700))
+                return NULL;
+        return template;
+}
+
+
diff -ur cron-3.0pl1.orig-solaris/compat.h cron-3.0pl1.orig/compat.h
--- cron-3.0pl1.orig-solaris/compat.h	2009-03-12 13:30:53.000000000 +0000
+++ cron-3.0pl1.orig/compat.h	2009-03-12 15:20:37.912499900 +0000
@@ -63,7 +63,8 @@
 
 #ifndef POSIX
 # if (BSD >= 199103) || defined(__linux__) || defined(__GNU__) || defined(ultrix) ||\
-        defined(AIX) ||\ defined(HPUX) || defined(CONVEX) || defined(IRIX) || defined(__GLIBC__)
+        defined(AIX) ||\
+defined(HPUX) || defined(CONVEX) || defined(IRIX) || defined(__GLIBC__)
 #  define POSIX
 # endif
 #endif
@@ -86,7 +87,7 @@
 #endif
 
 #if (!defined(BSD) || (BSD < 198911)) && !defined(__linux__) &&\
-	!defined(IRIX) && !defined(UNICOS) && !defined(HPUX) && !defined(__GNU__) && !defined(__GLIBC__)
+	!defined(IRIX) && !defined(UNICOS) && !defined(HPUX) && !defined(__GNU__) && !defined(__GLIBC__) && !defined(__SUNPRO_C)
 # define NEED_STRDUP
 #endif
 
@@ -110,7 +111,7 @@
 # define HAVE_SAVED_UIDS
 #endif
 
-#if !defined(ATT) && !defined(__linux__) && !defined(__GNU__) && !defined(IRIX) && !defined(UNICOS) && !defined(__GLIBC__)
+        #if !defined(ATT) && !defined(__linux__) && !defined(__GNU__) && !defined(IRIX) && !defined(UNICOS) && !defined(__GLIBC__) && !defined(__SUNPRO_C)
 # define USE_SIGCHLD
 #endif
 
Only in cron-3.0pl1.orig: compat.o
Only in cron-3.0pl1.orig: cron
diff -ur cron-3.0pl1.orig-solaris/cron.c cron-3.0pl1.orig/cron.c
--- cron-3.0pl1.orig-solaris/cron.c	2009-03-12 13:30:53.000000000 +0000
+++ cron-3.0pl1.orig/cron.c	2009-03-12 15:09:25.251653050 +0000
@@ -401,7 +401,7 @@
 
 #ifdef USE_SIGCHLD
 static void
-sigchld_handler(x) {
+sigchld_handler(int x) {
 	int save_errno = errno;
 	WAIT_T		waiter;
 	PID_T		pid;
@@ -435,7 +435,7 @@
 
 
 static void
-sighup_handler(x) {
+sighup_handler(int x) {
 	log_close();
 
 	/* we should use sigaction for proper signal blocking as this 
diff -ur cron-3.0pl1.orig-solaris/cron.h cron-3.0pl1.orig/cron.h
--- cron-3.0pl1.orig-solaris/cron.h	2009-03-12 13:30:53.000000000 +0000
+++ cron-3.0pl1.orig/cron.h	2009-03-12 15:19:14.122628700 +0000
@@ -32,7 +32,7 @@
 
 #include <stdio.h>
 #include <ctype.h>
-#include <bitstring.h>
+#include "bitstring.h"
 #include <pwd.h>
 #include <sys/wait.h>
 
Only in cron-3.0pl1.orig: cron.o
Only in cron-3.0pl1.orig: crontab
Only in cron-3.0pl1.orig: crontab.o
Only in cron-3.0pl1.orig: database.o
Only in cron-3.0pl1.orig: do_command.o
Only in cron-3.0pl1.orig: entry.o
Only in cron-3.0pl1.orig: env.o
diff -ur cron-3.0pl1.orig-solaris/externs.h cron-3.0pl1.orig/externs.h
--- cron-3.0pl1.orig-solaris/externs.h	2009-03-12 13:30:53.000000000 +0000
+++ cron-3.0pl1.orig/externs.h	2009-03-12 13:35:46.000000000 +0000
@@ -116,11 +116,11 @@
 #endif
 
 #ifdef NEED_STRCASECMP
-extern	int		strcasecmp __P((char *, char *));
+extern	int		strcasecmp __P((const char *, const char *));
 #endif
 
 #ifdef NEED_STRDUP
-extern	char		*strdup __P((char *));
+extern	char		*strdup __P((const char *));
 #endif
 
 #ifdef NEED_STRERROR
Only in cron-3.0pl1.orig: job.o
diff -ur cron-3.0pl1.orig-solaris/Makefile cron-3.0pl1.orig/Makefile
--- cron-3.0pl1.orig-solaris/Makefile	2009-03-12 13:30:53.000000000 +0000
+++ cron-3.0pl1.orig/Makefile	2009-03-12 17:39:22.971233150 +0000
@@ -47,7 +47,7 @@
 
 #################################### begin configurable stuff
 #<<DESTROOT is assumed to have ./etc, ./bin, and ./man subdirectories>>
-DESTROOT	=	$(DESTDIR)/usr
+DESTROOT	=	$(DESTDIR)/opt/csw
 DESTSBIN	=	$(DESTROOT)/sbin
 DESTBIN		=	$(DESTROOT)/bin
 DESTMAN		=	$(DESTROOT)/share/man
@@ -57,7 +57,7 @@
 #<<need getopt()>>
 LIBS		= $(PAM_LIBS) $(SELINUX_LIBS) $(AUDIT_LIBS)
 #<<optimize or debug?>>
-OPTIM		=	-O2
+#OPTIM		=	-O2
 #OPTIM		=	-g
 #<<ATT or BSD or POSIX?>>
 # (ATT untested)
@@ -74,12 +74,12 @@
 # Allow override from command line
 DEBUG_DEFS = -DDEBUGGING=0   
 # The -DUSE_SIGCHLD is needed for the Alpha port
-DEFS = -DDEBIAN -DUSE_SIGCHLD $(DEBUG_DEFS) $(PAM_DEFS) $(SELINUX_DEFS)
+#DEFS = -DDEBIAN -DUSE_SIGCHLD $(DEBUG_DEFS) $(PAM_DEFS) $(SELINUX_DEFS)
 #(SGI IRIX systems need this)
 #DEFS		=	-D_BSD_SIGNALS -Dconst=
 #<<the name of the BSD-like install program>>
 #INSTALL = installbsd
-INSTALL = install -s
+INSTALL = ginstall -s
 #<<any special load flags>>
 # LDFLAGS		=	-s
 # Let install do the strip
@@ -87,7 +87,7 @@
 #################################### end configurable stuff
 
 SHELL		=	/bin/sh
-CFLAGS		=	$(OPTIM) $(INCLUDE) $(COMPAT) $(DEFS)
+# CFLAGS		=	$(OPTIM) $(INCLUDE) $(COMPAT) $(DEFS)
 
 INFOS		=	README CHANGES FEATURES INSTALL CONVERSION THANKS MAIL
 MANPAGES	=	bitstring.3 crontab.5 crontab.1 cron.8 putman.sh
@@ -118,8 +118,8 @@
 			$(CC) $(LDFLAGS) -o crontab $(CRONTAB_OBJ) $(LIBS)
 
 install		:	all
-			$(INSTALL) -c -m  755 -o root cron    $(DESTSBIN)/
-			$(INSTALL) -c -m 4755 -o root crontab $(DESTBIN)/
+			$(INSTALL) -c -m  755 cron    $(DESTSBIN)/
+			$(INSTALL) -c -m 2755 crontab $(DESTBIN)/
 			sh putman.sh crontab.1 $(DESTMAN)
 			sh putman.sh cron.8    $(DESTMAN)
 			sh putman.sh crontab.5 $(DESTMAN)
diff -ur cron-3.0pl1.orig-solaris/misc.c cron-3.0pl1.orig/misc.c
--- cron-3.0pl1.orig-solaris/misc.c	2009-03-12 13:30:53.000000000 +0000
+++ cron-3.0pl1.orig/misc.c	2009-03-12 13:38:43.000000000 +0000
@@ -42,6 +42,7 @@
 # include <syslog.h>
 #endif
 
+#include <sys/file.h>
 
 #if defined(LOG_DAEMON) && !defined(LOG_CRON)
 #define LOG_CRON LOG_DAEMON
@@ -277,7 +278,8 @@
 			exit(ERROR_EXIT);
 		}
 
-		if (flock(fd, LOCK_EX|LOCK_NB) < OK) {
+                /* This bit originally used flock(), switched to lockf() */
+		if (lockf(fd, F_TLOCK, 0) < OK) {
 			int save_errno = errno;
 
 			fscanf(fp, "%d", &otherpid);
Only in cron-3.0pl1.orig: misc.o
diff -ur cron-3.0pl1.orig-solaris/popen.c cron-3.0pl1.orig/popen.c
--- cron-3.0pl1.orig-solaris/popen.c	2009-03-12 13:30:53.000000000 +0000
+++ cron-3.0pl1.orig/popen.c	2009-03-12 16:04:25.829649200 +0000
@@ -35,6 +35,10 @@
 #  include <grp.h>
 #endif
 
+#include<string.h>
+#include<strings.h>
+#include<sys/types.h>
+#include<sys/stat.h>
 
 #define MAX_ARGS 100
 #define WANT_GLOBBING 0
@@ -167,9 +171,12 @@
 	FILE *iop;
 {
 	register int fdes;
-	int omask;
 	WAIT_T stat_loc;
 	PID_T pid;
+        /*
+         * http://www.phys.ufl.edu/docs/porting-faq.html
+         */
+        sigset_t block, oblock;
 
 	/*
 	 * pclose returns -1 if stream is not associated with a
@@ -178,10 +185,20 @@
 	if (pids == 0 || pids[fdes = fileno(iop)] == 0)
 		return(-1);
 	(void)fclose(iop);
-	omask = sigblock(sigmask(SIGINT)|sigmask(SIGQUIT)|sigmask(SIGHUP));
+        /*
+         * omask = sigblock(sigmask(SIGINT)|sigmask(SIGQUIT)|sigmask(SIGHUP));
+         */
+        (void)sigemptyset(&block);
+        (void)sigaddset(&block, SIGINT | SIGQUIT | SIGHUP);
+        if (sigprocmask(SIG_BLOCK, &block, &oblock) < 0)
+                log_it("CRON", getpid(), "error", "sigprocmask")
+                ;
 	while ((pid = wait(&stat_loc)) != pids[fdes] && pid != -1)
 		;
-	(void)sigsetmask(omask);
+        /*
+         * (void)sigsetmask(omask);
+         */
+        (void)sigprocmask(SIG_SETMASK, &oblock, (sigset_t *)NULL);
 	pids[fdes] = 0;
 	return (pid == -1 ? -1 : WEXITSTATUS(stat_loc));
 }
Only in cron-3.0pl1.orig: popen.o
diff -ur cron-3.0pl1.orig-solaris/putman.sh cron-3.0pl1.orig/putman.sh
--- cron-3.0pl1.orig-solaris/putman.sh	1994-09-01 20:17:17.000000000 +0000
+++ cron-3.0pl1.orig/putman.sh	2009-03-12 20:04:07.194853350 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 # putman.sh - install a man page according to local custom
 # vixie 27dec93 [original]
Only in cron-3.0pl1.orig: user.o
