diff -ru portage-2.1.6.orig/bin/archive-conf portage-2.1.6/bin/archive-conf
--- portage-2.1.6.orig/bin/archive-conf	2009-03-21 00:28:52.000000000 +0000
+++ portage-2.1.6/bin/archive-conf	2009-03-21 00:54:44.117069233 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 # Copyright 1999-2006 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id: archive-conf 12907 2009-03-11 05:47:00Z zmedico $
diff -ru portage-2.1.6.orig/bin/chkcontents portage-2.1.6/bin/chkcontents
--- portage-2.1.6.orig/bin/chkcontents	2007-02-10 21:43:35.000000000 +0000
+++ portage-2.1.6/bin/chkcontents	2009-03-21 00:54:48.499658975 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 # Copyright 1999-2006 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id: chkcontents 5945 2007-02-10 21:43:35Z zmedico $
diff -ru portage-2.1.6.orig/bin/clean_locks portage-2.1.6/bin/clean_locks
--- portage-2.1.6.orig/bin/clean_locks	2007-02-10 21:43:35.000000000 +0000
+++ portage-2.1.6/bin/clean_locks	2009-03-21 00:54:48.585042693 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python -O
+#!/usr/bin/env python -O
 # Copyright 1999-2006 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id: clean_locks 5945 2007-02-10 21:43:35Z zmedico $
diff -ru portage-2.1.6.orig/bin/dispatch-conf portage-2.1.6/bin/dispatch-conf
--- portage-2.1.6.orig/bin/dispatch-conf	2009-03-21 00:28:52.000000000 +0000
+++ portage-2.1.6/bin/dispatch-conf	2009-03-21 00:54:43.897127741 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python -O
+#!/usr/bin/env python -O
 # Copyright 1999-2006 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id: dispatch-conf 12219 2008-12-12 21:24:53Z zmedico $
diff -ru portage-2.1.6.orig/bin/ebuild portage-2.1.6/bin/ebuild
--- portage-2.1.6.orig/bin/ebuild	2009-03-21 00:28:52.000000000 +0000
+++ portage-2.1.6/bin/ebuild	2009-03-21 00:54:44.494618434 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python -O
+#!/usr/bin/env python -O
 # Copyright 1999-2006 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id: ebuild 12956 2009-03-11 06:16:25Z zmedico $
diff -ru portage-2.1.6.orig/bin/ebuild-helpers/dohtml portage-2.1.6/bin/ebuild-helpers/dohtml
--- portage-2.1.6.orig/bin/ebuild-helpers/dohtml	2009-03-21 00:28:52.000000000 +0000
+++ portage-2.1.6/bin/ebuild-helpers/dohtml	2009-03-21 00:54:45.253499181 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 # Copyright 1999-2006 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id: dohtml 13063 2009-03-12 00:28:45Z zmedico $
diff -ru portage-2.1.6.orig/bin/ebuild.sh portage-2.1.6/bin/ebuild.sh
--- portage-2.1.6.orig/bin/ebuild.sh	2009-03-21 00:28:52.000000000 +0000
+++ portage-2.1.6/bin/ebuild.sh	2009-03-21 15:51:43.037193320 +0000
@@ -1706,7 +1706,7 @@
 if [ -n "${EBUILD_SH_ARGS}" ] && \
 	! hasq ${EBUILD_SH_ARGS} clean depend help info nofetch ; then
 
-	if [ "$(id -nu)" == "portage" ] ; then
+	if [ "$(/opt/csw/bin/gid -nu)" == "portage" ] ; then
 		export USER=portage
 	fi
 
diff -ru portage-2.1.6.orig/bin/emaint portage-2.1.6/bin/emaint
--- portage-2.1.6.orig/bin/emaint	2008-11-24 01:41:15.000000000 +0000
+++ portage-2.1.6/bin/emaint	2009-03-21 00:54:47.755411952 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python -O
+#!/usr/bin/env python -O
 # vim: noet :
 import os
 import re
diff -ru portage-2.1.6.orig/bin/emerge portage-2.1.6/bin/emerge
--- portage-2.1.6.orig/bin/emerge	2009-03-21 00:28:52.000000000 +0000
+++ portage-2.1.6/bin/emerge	2009-03-21 00:54:47.795937579 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 # Copyright Gentoo Foundation 2006
 # Portage Emerge bits
 # $Id: emerge 13046 2009-03-11 07:44:10Z zmedico $
diff -ru portage-2.1.6.orig/bin/env-update portage-2.1.6/bin/env-update
--- portage-2.1.6.orig/bin/env-update	2007-02-10 21:43:35.000000000 +0000
+++ portage-2.1.6/bin/env-update	2009-03-21 00:54:48.171797217 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python -O
+#!/usr/bin/env python -O
 # Copyright 1999-2006 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id: env-update 5945 2007-02-10 21:43:35Z zmedico $
diff -ru portage-2.1.6.orig/bin/fixpackages portage-2.1.6/bin/fixpackages
--- portage-2.1.6.orig/bin/fixpackages	2007-05-31 06:43:41.000000000 +0100
+++ portage-2.1.6/bin/fixpackages	2009-03-21 00:54:48.735265590 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 # Copyright 1999-2006 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id: fixpackages 6687 2007-05-31 05:43:41Z zmedico $
diff -ru portage-2.1.6.orig/bin/md5check.py portage-2.1.6/bin/md5check.py
--- portage-2.1.6.orig/bin/md5check.py	2008-03-30 20:20:46.000000000 +0100
+++ portage-2.1.6/bin/md5check.py	2009-03-21 00:54:47.537861287 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python -O
+#!/usr/bin/env python -O
 # Copyright 1999-2006 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id: md5check.py 9632 2008-03-30 19:20:46Z zmedico $
diff -ru portage-2.1.6.orig/bin/pemerge.py portage-2.1.6/bin/pemerge.py
--- portage-2.1.6.orig/bin/pemerge.py	2006-06-10 22:40:40.000000000 +0100
+++ portage-2.1.6/bin/pemerge.py	2009-03-21 00:54:48.318275271 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python -O
+#!/usr/bin/env python -O
 
 import profile,time,sys,os
 sys.path = ["/usr/lib/portage/bin","/usr/lib/portage/pym"]+sys.path
diff -ru portage-2.1.6.orig/bin/pkgname portage-2.1.6/bin/pkgname
--- portage-2.1.6.orig/bin/pkgname	2008-03-15 22:43:25.000000000 +0000
+++ portage-2.1.6/bin/pkgname	2009-03-21 00:54:48.059457317 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 # Copyright 1999-2006 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id: pkgname 9470 2008-03-15 22:43:25Z genone $
diff -ru portage-2.1.6.orig/bin/portageq portage-2.1.6/bin/portageq
--- portage-2.1.6.orig/bin/portageq	2008-11-14 04:46:12.000000000 +0000
+++ portage-2.1.6/bin/portageq	2009-03-21 00:54:48.422413695 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python -O
+#!/usr/bin/env python -O
 # Copyright 1999-2006 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id: portageq 11895 2008-11-14 04:46:12Z zmedico $
diff -ru portage-2.1.6.orig/bin/quickpkg portage-2.1.6/bin/quickpkg
--- portage-2.1.6.orig/bin/quickpkg	2009-03-21 00:28:53.000000000 +0000
+++ portage-2.1.6/bin/quickpkg	2009-03-21 00:54:43.968195977 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 # Copyright 1999-2007 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id: quickpkg 13123 2009-03-17 19:13:33Z zmedico $
diff -ru portage-2.1.6.orig/bin/regenworld portage-2.1.6/bin/regenworld
--- portage-2.1.6.orig/bin/regenworld	2008-11-01 03:14:47.000000000 +0000
+++ portage-2.1.6/bin/regenworld	2009-03-21 00:54:44.774080845 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 # Copyright 1999-2006 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id: regenworld 11771 2008-11-01 03:14:47Z zmedico $
diff -ru portage-2.1.6.orig/bin/repoman portage-2.1.6/bin/repoman
--- portage-2.1.6.orig/bin/repoman	2009-03-21 00:28:53.000000000 +0000
+++ portage-2.1.6/bin/repoman	2009-03-21 00:54:47.362500028 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python -O
+#!/usr/bin/env python -O
 # Copyright 1999-2006 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id: repoman 13125 2009-03-17 19:15:21Z zmedico $
diff -ru portage-2.1.6.orig/bin/xpak portage-2.1.6/bin/xpak
--- portage-2.1.6.orig/bin/xpak	2007-02-10 21:43:35.000000000 +0000
+++ portage-2.1.6/bin/xpak	2009-03-21 00:54:45.033773639 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python -O
+#!/usr/bin/env python -O
 # Copyright 1999-2006 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id: xpak 5945 2007-02-10 21:43:35Z zmedico $
diff -ru portage-2.1.6.orig/pym/_emerge/__init__.py portage-2.1.6/pym/_emerge/__init__.py
--- portage-2.1.6.orig/pym/_emerge/__init__.py	2009-03-21 00:28:53.000000000 +0000
+++ portage-2.1.6/pym/_emerge/__init__.py	2009-03-21 11:44:34.532033847 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python -O
+#!/usr/bin/env python -O
 # Copyright 1999-2006 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id: __init__.py 13127 2009-03-17 19:30:58Z zmedico $
@@ -12284,9 +12284,9 @@
 				except OSError:
 					pass
 			if stat.S_ISDIR(mymode):
-				mycommand = "find '%s' -name '.*' -type d -prune -o -name '._cfg????_*'" % x
+				mycommand = "gfind '%s' -name '.*' -type d -prune -o -name '._cfg????_*'" % x
 			else:
-				mycommand = "find '%s' -maxdepth 1 -name '._cfg????_%s'" % \
+				mycommand = "gfind '%s' -maxdepth 1 -name '._cfg????_%s'" % \
 					os.path.split(x.rstrip(os.path.sep))
 			mycommand += " ! -name '.*~' ! -iname '.*.bak' -print0"
 			a = commands.getstatusoutput(mycommand)
@@ -12457,8 +12457,8 @@
 				"control (contains %s).\n!!! Aborting rsync sync.\n") % \
 				(myportdir, vcs_dir), level=logging.ERROR, noiselevel=-1)
 			return 1
-		if not os.path.exists("/usr/bin/rsync"):
-			print "!!! /usr/bin/rsync does not exist, so rsync support is disabled."
+		if not os.path.exists("/opt/csw/bin/rsync"):
+			print "!!! /opt/csw/bin/rsync does not exist, so rsync support is disabled."
 			print "!!! Type \"emerge net-misc/rsync\" to enable rsync support."
 			sys.exit(1)
 		mytimeout=180
@@ -12629,7 +12629,7 @@
 			if mytimestamp != 0 and "--quiet" not in myopts:
 				print ">>> Checking server timestamp ..."
 
-			rsynccommand = ["/usr/bin/rsync"] + rsync_opts + extra_rsync_opts
+			rsynccommand = ["/opt/csw/bin/rsync"] + rsync_opts + extra_rsync_opts
 
 			if "--debug" in myopts:
 				print rsynccommand
diff -ru portage-2.1.6.orig/pym/portage/tests/runTests portage-2.1.6/pym/portage/tests/runTests
--- portage-2.1.6.orig/pym/portage/tests/runTests	2009-03-21 00:28:53.000000000 +0000
+++ portage-2.1.6/pym/portage/tests/runTests	2009-03-21 00:54:55.109264983 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 # runTests.py -- Portage Unit Test Functionality
 # Copyright 2006 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
diff -ru portage-2.1.6.orig/src/python-missingos/setup.py portage-2.1.6/src/python-missingos/setup.py
--- portage-2.1.6.orig/src/python-missingos/setup.py	2006-06-10 22:40:40.000000000 +0100
+++ portage-2.1.6/src/python-missingos/setup.py	2009-03-21 00:54:49.112474798 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 # $Id: setup.py 3483 2006-06-10 21:40:40Z genone $
 
 from os import chdir, stat
